lib/Semantics/assignment.cpp

```diff
@@ -43,7 +43,7 @@ class AssignmentContext {
   void Analyze(const parser::ConcurrentControl &);
 
 private:
-  void CheckForPureContext(const SomeExpr &lhs, const SomeExpr &rhs,
+  bool CheckForPureContext(const SomeExpr &lhs, const SomeExpr &rhs,
```

Comment:
This function name is not that evocative of its return value.  It would be nice to have a name that gave an indication of the meaning of the return result.  How about "AssignableInPureContext()"?

Reply:
That's an adjective.